systems/src/apu/mod.rs

```diff
@@ -0,0 +1,1396 @@
+                AMBIENT_TEMPERATURE
+            );
+        }
+
+        #[test]
+        fn when_apu_starting_egt_reaches_above_700_degree_celsius() {
+            let mut tester = tester_with().starting_apu();
+            let mut max_egt: f64 = 0.;
+
+            loop {
+                tester = tester.run(Duration::from_secs(1));
+
+                let egt = tester.get_egt().get::<degree_celsius>();
+                if egt < max_egt {
+                    break;
+                }
+
+                max_egt = egt;
+            }
+
+            assert!(max_egt > 700.);
+        }
+
+        #[test]
+        fn egt_max_always_33_above_egt_warn() {
+            let mut tester = tester_with().starting_apu();
+
+            for _ in 1..=100 {
+                tester = tester.run(Duration::from_secs(1));
+
+                assert_about_eq!(
+                    tester.get_egt_warning_temperature().get::<degree_celsius>(),
+                    tester.get_egt_caution_temperature().get::<degree_celsius>() + 33.
+                );
+            }
+        }
+
+        #[test]
+        fn start_sw_on_light_turns_off_when_apu_available() {
+            let mut tester = tester_with().starting_apu();
+
+            loop {
+                tester = tester.run(Duration::from_secs(1));
+
+                if tester.apu_is_available() {
+                    break;
+                }
+            }
+
+            assert!(!tester.start_is_on());
+            assert!(tester.start_shows_available());
+        }
+
+        #[test]
+        fn when_apu_bleed_valve_open_on_shutdown_cooldown_period_commences_and_apu_remains_available(
+        ) {
+            // The cool down period is between 60 to 120. It is configurable by aircraft mechanics and
+            // we'll make it a configurable option in the sim. For now, 120s.
+            let mut tester =
+                tester_with()
+                    .running_apu()
+                    .and()
+                    .master_off()
+                    .run(Duration::from_millis(
+                        ElectronicControlBox::BLEED_AIR_COOLDOWN_DURATION_MILLIS,
+                    ));
+
+            assert!(tester.apu_is_available());
+
+            // Move from Running to Shutdown turbine state.
+            tester = tester.run(Duration::from_millis(1));
+            // APU N reduces below 95%.
+            tester = tester.run(Duration::from_secs(5));
+
+            assert!(!tester.apu_is_available());
+        }
+
+        #[test]
+        fn when_shutting_down_apu_remains_available_until_n_less_than_95() {
+            let mut tester = tester_with()
+                .running_apu_without_bleed_air()
+                .and()
+                .master_off();
+
+            let mut n = 100.;
+
+            while 0. < n {
+                tester = tester.run(Duration::from_millis(50));
+                n = tester.get_n().get::<percent>();
+                assert_eq!(tester.apu_is_available(), 95. <= n);
```

Comment:
I guess this test doesn't explicitly ensure you ever hit the N < 95 case. Maybe add assertions that you hit both cases at least once? I.e. one before, and one after.

Reply:
I've added `assert!(tester.apu_is_available());` before it. There's no need to add an assert after it, because the assertion always executes after the tester ran; even when N hit 0 it hits the assertion once.